### PR TITLE
Report deleted/inserted content in Chrome

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
None.

### Summary of the issue:
Web authors are able to mark particular content as being inserted or deleted with ins and del tags respectively.
The IAccessible2 spec has been updated to include roles for ins and del, and Chrome Canary now exposes these roles.  NVDA should add its own roles for these, mapped from the IAccessible2 roles. 
 
### Description of how this pull request fixes the issue:
* NVDA now has deleted_content and inserted_content roles. 
* These are appropriately reported in focus mode and browseMode.
* The new IAccessible2 content_deletion and content_insertion roles are mapped to the added NVDA roles, thus making this information accessible in Google Chrome.
* Braille also has shortened role labels of "ins" and "del".

### Testing performed:
Tested 
[marked text.html.txt](https://github.com/nvaccess/nvda/files/2229650/marked.text.html.txt)
 containing ins / del tags in Google Chrome Canary. 
NVDA reports "inserted" and "deleted" for the ins and del tags respectively.

### Known issues with pull request:
None.

### Change log entry:
New features:
* In Google Chrome, the existance of inserted and deleted content is now reported.

### CODE REVIEW

**Comment by leonardder** [comment]
Have you considered consistency here? We now have two separate insertion/deletion implementations, one based on control field attributes (Word and UIA) and one based on roles (IA2). It would be great if we could choose one of them, but I assume that introduces othher issues??

File: source/controlTypes.py [hunk 1]
```diff
@@ -486,6 +488,10 @@
 	ROLE_CHARTELEMENT:_("chart element"),
```
Comment (leonardder) [inline source/controlTypes.py]: Could you also move this one below role_audio?

**Comment by michaelDCurran** [comment]
Yes, there is one implementation for format fields, and one for roles / 
controlFields.

Microsoft word exposes a flat view of text, where revisions are 
assentially a format change. Web browsers such as Chrome expose a tree 
structure, where revisions (such as insertion/deletion) are just one of 
the nodes in the tree.
We could try and massage one of them into the other, but I don't think 
it is worth the complication.
At the end of the day they are represented topologically very different  
in each application. Abstraction can only go so far.
I suppose you *could* argue that we should change our mshtml 
virtualBuffer to match Chrome... currently we are somehow faking ins/del 
as formatFields. But, I'm again not sure it is worth the code chirn, 
especially for an unmaintained browser.

**Comment by Brian1Gaff** [comment]
I wonder if any of this has a knock on effect on software which uses mshtml 
like some email clients still do?
 Brian

File: source/textInfos/__init__.py [hunk 2]
```diff
@@ -68,12 +68,13 @@ def getPresentationCategory(self, ancestors, formatConfig, reason=controlTypes.R
...
 
 		if (
-			role in (controlTypes.ROLE_LINK, controlTypes.ROLE_HEADING, controlTypes.ROLE_BUTTON, controlTypes.ROLE_RADIOBUTTON, controlTypes.ROLE_CHECKBOX, controlTypes.ROLE_GRAPHIC, controlTypes.ROLE_CHART, controlTypes.ROLE_MENUITEM, controlTypes.ROLE_TAB, controlTypes.ROLE_COMBOBOX, controlTypes.ROLE_SLIDER, controlTypes.ROLE_SPINBUTTON, controlTypes.ROLE_COMBOBOX, controlTypes.ROLE_PROGRESSBAR, controlTypes.ROLE_TOGGLEBUTTON, controlTypes.ROLE_MENUBUTTON, controlTypes.ROLE_TREEVIEW, controlTypes.ROLE_CHECKMENUITEM, controlTypes.ROLE_RADIOMENUITEM)
+			role in (controlTypes.ROLE_DELETED_CONTENT,controlTypes.ROLE_INSERTED_CONTENT,controlTypes.ROLE_LINK, controlTypes.ROLE_HEADING, controlTypes.ROLE_BUTTON, controlTypes.ROLE_RADIOBUTTON, controlTypes.ROLE_CHECKBOX, controlTypes.ROLE_GRAPHIC, controlTypes.ROLE_CHART, controlTypes.ROLE_MENUITEM, controlTypes.ROLE_TAB, controlTypes.ROLE_COMBOBOX, controlTypes.ROLE_SLIDER, controlTypes.ROLE_SPINBUTTON, controlTypes.ROLE_COMBOBOX, controlTypes.ROLE_PROGRESSBAR, controlTypes.ROLE_TOGGLEBUTTON, controlTypes.ROLE_MENUBUTTON, controlTypes.ROLE_TREEVIEW, controlTypes.ROLE_CHECKMENUITEM, controlTypes.ROLE_RADIOMENUITEM)
```
Comment (leonardder) [inline source/textInfos/__init__.py]: Could you split this into multiple lines?

**Review by LeonarddeR — DISMISSED** [review]

**Review by LeonarddeR — APPROVED** [review]

**Pull request merged**